scripts/msmarco_v2_doc_segment.py

```diff
@@ -0,0 +1,90 @@
+import argparse
+import os
+import sys
+import gzip
+import json
+import spacy #Currently using spacy 2.3.5
+from tqdm import tqdm
+import re
+import glob
+from multiprocessing import Pool
+
+def create_segments(doc_text, max_length, stride):
+    doc_text = doc_text.strip()
+    doc = nlp(doc_text[:10000])
+    sentences = [sent.string.strip() for sent in doc.sents]
+    segments = []
+    
+    for i in range(0, len(sentences), stride):
+        segment = " ".join(sentences[i:i+max_length])
+        segments.append(segment)
+        if i + max_length >= len(sentences):
+            break
+    return segments
+
+def split_document(f_ins, f_out): 
+    print('Spliting documents...')
+    output = open(f_out, 'w')
+    output_id = open(f_out.replace(".json", ".id"), 'w')
+    for f_in in f_ins:
+        with gzip.open(f_in, 'rt', encoding='utf8') as in_fh:
+            for json_string in tqdm(in_fh):
+                doc = json.loads(json_string)
+                f_doc_id = doc['docid']
+                doc_url = doc['url']
+                doc_title = doc['title']
+                doc_headings = doc['headings']
+                doc_text = doc['body']
+
+                segments = create_segments(doc_text, args.max_length, args.stride)
+
+                for seg_id, segment in enumerate(segments):
+                    # expanded_text = f'{doc_url}\n{doc_headings}\n{doc_title}\n{segment}'
+                    doc_seg = f'{f_doc_id}#{seg_id}'
+                    output_dict = {'docid': doc_seg, 'url': doc_url, 'title': doc_title, 'headings': doc_headings, 'segment': segment}
+                    output.write(json.dumps(output_dict) + '\n')  
+                    output_id.write(doc_seg+'\n')
+
+    output.close()
+    output_id.close()
+    print('Done!')
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser(
+        description='Concatenate MS MARCO original docs with predicted queries')
+    parser.add_argument('--original_docs_path', required=True, help='MS MARCO .tsv corpus file.')
+    parser.add_argument('--output_docs_path', required=True, help='Output file in the anserini jsonl format.')
+    parser.add_argument('--max_length', default=10)
```

Comment:
`--max-length`? hyphen instead of underscore.

Reply:
OK. Not sure shall we align this rule ? I found other scripts use underscore (https://github.com/castorini/pyserini/blob/master/scripts/entity_linking.py)